docs/configuration.v2alpha1.md

```diff
@@ -147,15 +149,199 @@ spec:
+| [key].createRbac | Set CreateRbac to false to prevent automatic creation of Role/ClusterRole for this component |
+| [key].customConfigurations `map[string]object` | CustomConfiguration allows to specify custom configuration files for `datadog.yaml`, `datadog-cluster.yaml`, `security-agent.yaml`, and `system-probe.yaml`. The content will be merged with configuration generated by the Datadog Operator, with priority given to custom configuration. WARNING: It's thus possible to override values set in the `DatadogAgent`. |
+| [key].customConfigurations.[key].configData | ConfigData corresponds to the configuration file content. |
+| [key].customConfigurations.[key].configMap.items | Items maps a ConfigMap data key to a file path mount. |
+| [key].customConfigurations.[key].configMap.name | Name is the name of the ConfigMap. |
+| [key].disabled | Disabled force disables a component. |
+| [key].env `[]object` | Specify additional environmental variables for all containers in this component Priority is Container > Component See also: https://docs.datadoghq.com/agent/kubernetes/?tab=helm#environment-variables |
+| [key].extraChecksd.configDataMap | ConfigDataMap corresponds to the content of the configuration files. They key should be the filename the contents get mounted to; for instance check.py or check.yaml. |
+| [key].extraChecksd.configMap.items | Items maps a ConfigMap data key to a file path mount. |
+| [key].extraChecksd.configMap.name | Name is the name of the ConfigMap. |
+| [key].extraConfd.configDataMap | ConfigDataMap corresponds to the content of the configuration files. They key should be the filename the contents get mounted to; for instance check.py or check.yaml. |
+| [key].extraConfd.configMap.items | Items maps a ConfigMap data key to a file path mount. |
+| [key].extraConfd.configMap.name | Name is the name of the ConfigMap. |
+| [key].hostNetwork | Host networking requested for this pod. Use the host's network namespace. |
+| [key].hostPID | Use the host's pid namespace. |
+| [key].image.jmxEnabled | Define whether the Agent image should support JMX. To be used if the Name field does not correspond to a full image string. |
+| [key].image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7. Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7. Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent. Use "agent" with the registry and tag configurations for <registry>/agent:<tag>. Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>. If the name is full image string - `<name>:<tag>` or `<registry>/<name>:<tag>`, `tag` and `jmxEnabled` values are ignored, so is `global.registry`. Otherwise, image string is created by overriding default settings with supplied `name`, `tag` and `jmxEnabled` values; image string is created using default registry unless `global.registry` is configured. |
```

Comment:
```suggestion
| [key].image.name | Define the image to use: Use "gcr.io/datadoghq/agent:latest" for Datadog Agent 7. Use "datadog/dogstatsd:latest" for standalone Datadog Agent DogStatsD 7. Use "gcr.io/datadoghq/cluster-agent:latest" for Datadog Cluster Agent. Use "agent" with the registry and tag configurations for <registry>/agent:<tag>. Use "cluster-agent" with the registry and tag configurations for <registry>/cluster-agent:<tag>. If the name is full image string - `<name>:<tag>` or `<registry>/<name>:<tag>`, then `tag`, `jmxEnabled`, and `global.registry` values are ignored. Otherwise, image string is created by overriding default settings with supplied `name`, `tag`, and `jmxEnabled` values; image string is created using default registry unless `global.registry` is configured. |
```